### PR TITLE
Fix Tests on Mac OS

### LINKED ISSUE #1046: You requested to install 'osgi.bundle; org.eclipse.jdt.launching.macosx 0.0.0' but it could not be found
When trying to execute `mvn clean verify` I get the following error on module `org.eclipse.m2e.tests.common`

```
...
[DEBUG] -- end configuration --
[DEBUG] Class org.eclipse.m2e.tests.common.AbstractLifecycleMappingTest matches the current filter.
[DEBUG] Class org.eclipse.m2e.tests.common.AbstractLifecycleMappingTest matches the current filter.
[DEBUG] No solution found because the problem is unsatisfiable.: [Unable to satisfy dependency from tycho-extra-1667843090263 0.0.0.1667843090263 to osgi.bundle; org.eclipse.jdt.launching.macosx 0.0.0.; Unable to satisfy dependency from tycho-1667843090269 0.0.0.1667843090269 to osgi.bundle; org.eclipse.jdt.launching.macosx 0.0.0.; No solution found because the problem is unsatisfiable.]
[INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=x86_64}
[ERROR] Cannot resolve project dependencies:
[ERROR]   You requested to install 'osgi.bundle; org.eclipse.jdt.launching.macosx 0.0.0' but it could not be found
[ERROR]
[ERROR] See https://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting for help.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  9.240 s
[INFO] Finished at: 2022-11-07T18:44:50+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-surefire-plugin:3.0.0:integration-test (default-integration-test) on project org.eclipse.m2e.tests.common: Execution default-integration-test of goal org.eclipse.tycho:tycho-surefire-plugin:3.0.0:integration-test failed: Cannot resolve dependencies of MavenProject: org.eclipse.m2e:org.eclipse.m2e.tests.common:2.0.2-SNAPSHOT @ /Users/konradwindszus/git/eclipse/m2e-core/org.eclipse.m2e.tests.common/.tycho-consumer-pom.xml: See log for details -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho:tycho-surefire-plugin:3.0.0:integration-test (default-integration-test) on project org.eclipse.m2e.tests.common: Execution default-integration-test of goal org.eclipse.tycho:tycho-surefire-plugin:3.0.0:integration-test failed: Cannot resolve dependencies of MavenProject: org.eclipse.m2e:org.eclipse.m2e.tests.common:2.0.2-SNAPSHOT @ /Users/konradwindszus/git/eclipse/m2e-core/org.eclipse.m2e.tests.common/.tycho-consumer-pom.xml
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:375)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-integration-test of goal org.eclipse.tycho:tycho-surefire-plugin:3.0.0:integration-test failed: Cannot resolve dependencies of MavenProject: org.eclipse.m2e:org.eclipse.m2e.tests.common:2.0.2-SNAPSHOT @ /Users/konradwindszus/git/eclipse/m2e-core/org.eclipse.m2e.tests.common/.tycho-consumer-pom.xml
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.tycho.DependencyResolutionException: Cannot resolve dependencies of MavenProject: org.eclipse.m2e:org.eclipse.m2e.tests.common:2.0.2-SNAPSHOT @ /Users/konradwindszus/git/eclipse/m2e-core/org.eclipse.m2e.tests.common/.tycho-consumer-pom.xml
    at org.eclipse.tycho.p2.resolver.P2ResolverImpl.resolveDependencies (P2ResolverImpl.java:253)
    at org.eclipse.tycho.p2.resolver.P2ResolverImpl.resolveTargetDependencies (P2ResolverImpl.java:130)
    at org.eclipse.tycho.p2.resolver.P2DependencyResolver.doResolveDependencies (P2DependencyResolver.java:427)
    at org.eclipse.tycho.p2.resolver.P2DependencyResolver.resolveDependencies (P2DependencyResolver.java:389)
    at org.eclipse.tycho.surefire.AbstractTestMojo.createEclipseInstallation (AbstractTestMojo.java:835)
    at org.eclipse.tycho.surefire.AbstractTestMojo.execute (AbstractTestMojo.java:704)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.tycho.p2.util.resolution.ResolverException: See log for details
    at org.eclipse.tycho.p2.util.resolution.ProjectorResolutionStrategy.resolve (ProjectorResolutionStrategy.java:129)
    at org.eclipse.tycho.p2.util.resolution.AbstractResolutionStrategy.resolve (AbstractResolutionStrategy.java:53)
    at org.eclipse.tycho.p2.resolver.P2ResolverImpl.resolveDependencies (P2ResolverImpl.java:236)
    at org.eclipse.tycho.p2.resolver.P2ResolverImpl.resolveTargetDependencies (P2ResolverImpl.java:130)
    at org.eclipse.tycho.p2.resolver.P2DependencyResolver.doResolveDependencies (P2DependencyResolver.java:427)
    at org.eclipse.tycho.p2.resolver.P2DependencyResolver.resolveDependencies (P2DependencyResolver.java:389)
    at org.eclipse.tycho.surefire.AbstractTestMojo.createEclipseInstallation (AbstractTestMojo.java:835)
    at org.eclipse.tycho.surefire.AbstractTestMojo.execute (AbstractTestMojo.java:704)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR]
```

I am using the following Maven version:
```
Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
Maven home: /opt/homebrew/Cellar/maven/3.8.6/libexec
Java version: 17.0.5, vendor: Eclipse Adoptium, runtime: /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home
Default locale: en_DE, platform encoding: UTF-8
OS name: "mac os x", version: "12.6.1", arch: "aarch64", family: "mac"
```

The dependency is declared in https://github.com/eclipse-m2e/m2e-core/blob/9416e758afe1a405fe6616ce110e7d974e73858e/pom.xml#L418 but it is not clear to me which repository is supposed to resolve this dependency.

### LINKED ISSUE #1080: Build and Test with GitHub Actions on Windows and Mac OS
Currently some tests fail on those OS which makes the life for contributors using one of those OS really hard.

This is related to #1046.

### PR DESCRIPTION
**Pull request by kwin**
This closes #1046
This closes #1080

### CODE REVIEW

**Comment by kwin** [comment]
@HannesWell For me the tests succeed on my local Mac now in this branch. Unfortunately I had to disable one test on my OS. Can you try it out with GH Action (just merge with #1081)

File: org.eclipse.m2e.core.tests/src/org/eclipse/m2e/core/internal/project/registry/MemoryConsumptionTest.java
Comment (HannesWell) [inline org.eclipse.m2e.core.tests/src/org/eclipse/m2e/core/internal/project/registry/MemoryConsumptionTest.java:74]: Couldn't you just use `Assert.assertTrue()`?

File: org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java
Comment (HannesWell) [inline org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java:221]: That looks like something that should not be platform depended and should always work.

File: pom.xml
Comment (HannesWell) [inline pom.xml:128]: IIRC this is discouraged.
Can you explain why this does not cause problems on Linux (where three out of the four envs don't match as well), but does on Mac?

**Review by HannesWell — COMMENTED** [review]
> @HannesWell For me the tests succeed on my local Mac now in this branch. Unfortunately I had to disable one test on my OS. Can you try it out with GH Action (just merge with #1081)

Sure. Just did it. Thanks for making it work. 👍🏽

File: org.eclipse.m2e.core.tests/src/org/eclipse/m2e/core/internal/project/registry/MemoryConsumptionTest.java
Comment (kwin) [reply org.eclipse.m2e.core.tests/src/org/eclipse/m2e/core/internal/project/registry/MemoryConsumptionTest.java:74]: Makes sense.

File: org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java
Comment (kwin) [reply org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java:221]: yes, this is a race condition which incidentally just happened on my Mac. IMHO there is a wait missing in the test case. @HannesWell Any idea how to wait for the termination?

File: pom.xml
Comment (kwin) [reply pom.xml:128]: There is a lengthy discussion around this in https://github.com/eclipse-m2e/m2e-core/discussions/1153. Probably a bug in Tycho, but I don't see an issue right now without environments....

File: org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java
Comment (laeubi) [reply org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java:221]: Waiting for launch termination is quite ugly, you need to register a listener and then signal that it is done e.g using a countdown latch.

Comment (kwin) [reply org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java:221]: Ok, can we just leave this assertion out from the test then?

Comment (laeubi) [reply org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java:221]: I assume there is a reason to check it is terminated, but I see some other problems here, for example that we assume there are always two launches and the first is always maven.

So I think the test might be improved in the following way:

1. Do not assert the number of launches, just assert that there are the two ones we expect (e.g. by searching for name or property)
2. If we assume they should terminate, check in a loop if they are terminated, if not do a Thread.spinWait() and check again, if not terminated after 10 seconds (or whatever seems suitable) give a meaningful exception.

Comment (kwin) [reply org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java:221]: @HannesWell You wrote the code initially, so I leave that up to you to decide.

File: org.eclipse.m2e.core.tests/src/org/eclipse/m2e/core/internal/project/registry/MemoryConsumptionTest.java
Comment (kwin) [reply org.eclipse.m2e.core.tests/src/org/eclipse/m2e/core/internal/project/registry/MemoryConsumptionTest.java]: Done in https://github.com/eclipse-m2e/m2e-core/pull/1154/commits/562182c8eb2bcdcdcf719012dd79c3aced3a69fd.

File: org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java
Comment (kwin) [reply org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java]: Fixed as suggested by @laeubi in https://github.com/eclipse-m2e/m2e-core/pull/1154/commits/562182c8eb2bcdcdcf719012dd79c3aced3a69fd.

File: pom.xml
Comment (kwin) [reply pom.xml]: As https://github.com/eclipse-tycho/tycho/issues/1900 has been fixed I restored the environments again.

Comment (HannesWell) [reply pom.xml]: Great. 👍🏽

File: org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java
Comment (HannesWell) [reply org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java]: In general I'm surprised that this is even an issue that the debugger-launch is not marked as terminated yet. At that point the maven build has completely terminated so the debugger attached to the forked surefire VM should have done the same for 'some' time too. But maybe you have a super fast computer or are just in bad luck to get hit by that race-condition.

> I assume there is a reason to check it is terminated, but I see some other problems here, for example that we assume there are always two launches and the first is always maven.
> 

It is indeed not very nice, but since the launches in the launch-manager are cleared before the build is started I think that is OK for a test-case and until now I never observed that as a problem. But of course it can be enhanced.

> Fixed as suggested by @laeubi in [562182c](https://github.com/eclipse-m2e/m2e-core/commit/562182c8eb2bcdcdcf719012dd79c3aced3a69fd).

Good. 👍🏽

File: pom.xml
Comment (HannesWell) [inline pom.xml:240]: Do you think that is necessary for the every-day use? Looks like a debug-option to me.

**Review by HannesWell — APPROVED** [review]
Looks good. Thank you.

I just have two minor remarks, but those are not critical.

And can you please either squash all commits into one and give them a meaningful clean-up message or squash-and-merge. IMHO there is no need to record all steps of the discussions in git. :)

I'll also try this change with https://github.com/eclipse-m2e/m2e-core/pull/1081.

File: org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java
Comment (HannesWell) [reply org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java]: Commited it via the GH web-interface.

**Comment by HannesWell** [comment]
The GH verification build for mac-os in https://github.com/eclipse-m2e/m2e-core/pull/1081 succeeded.

File: pom.xml
Comment (kwin) [reply pom.xml:240]: Agreed, I removed now.

**Comment by kwin** [comment]
@HannesWell As I don't have commit rights (yet), can you merge?

**Comment by HannesWell** [comment]
> @HannesWell As I don't have commit rights (yet), can you merge?

Sure. Thanks for fixing the mac build.

**Pull request merged**
